admin-dev/themes/new-theme/js/pages/order/view/order-product-cancel.js

```diff
@@ -0,0 +1,143 @@
+    this.orderDelivered = parseInt(this.cancelProductForm.data('isDelivered'), 10) === 1;
+    this.isTaxIncluded = parseInt(this.cancelProductForm.data('isTaxIncluded'), 10) === 1;
+    this.discountsAmount = parseFloat(this.cancelProductForm.data('discountsAmount'));
+    this.currencyFormatter = NumberFormatter.build(this.cancelProductForm.data('priceSpecification'));
+  }
+
+  showPartialRefund() {
+    // Always start by hiding elements then show the others, since some elements are common
+    $(OrderViewPageMap.cancelProduct.toggle.standardRefund).hide();
+    $(OrderViewPageMap.cancelProduct.toggle.partialRefund).show();
+    $(OrderViewPageMap.cancelProduct.table.actions).hide();
+    this.initForm(
+      $(OrderViewPageMap.cancelProduct.buttons.save).data('partialRefundLabel'),
+      this.router.generate('admin_orders_partial_refund', {orderId: this.orderId})
+    );
+  }
+
+  showStandardRefund() {
+    // Always start by hiding elements then show the others, since some elements are common
+    $(OrderViewPageMap.cancelProduct.toggle.partialRefund).hide();
+    $(OrderViewPageMap.cancelProduct.toggle.standardRefund).show();
+    $(OrderViewPageMap.cancelProduct.table.actions).hide();
+    this.initForm(
+      $(OrderViewPageMap.cancelProduct.buttons.save).data('standardRefundLabel'),
+      ''
+    );
+  }
+
+  hideRefund() {
+    $(OrderViewPageMap.cancelProduct.toggle.partialRefund).hide();
+    $(OrderViewPageMap.cancelProduct.toggle.standardRefund).hide();
+    $(OrderViewPageMap.cancelProduct.table.actions).show();
+  }
+
+  initForm(actionName, formAction) {
+    this.listenForInputs();
+    this.updateVoucherRefund();
+
+    this.cancelProductForm.attr('action', formAction);
+    $(OrderViewPageMap.cancelProduct.buttons.save).html(actionName);
+    $(OrderViewPageMap.cancelProduct.table.header).html(actionName);
+    $(OrderViewPageMap.cancelProduct.checkboxes.restock).attr('checked', this.orderDelivered);
+    $(OrderViewPageMap.cancelProduct.checkboxes.creditSlip).attr('checked', true);
+  }
+
+  listenForInputs() {
+    $(OrderViewPageMap.cancelProduct.inputs.quantity).off('change').on('change', (event) => {
+      const $productQuantityInput = $(event.target);
+      const $parentCell = $productQuantityInput.parents(OrderViewPageMap.cancelProduct.table.cell);
+      const $productAmount = $parentCell.find(OrderViewPageMap.cancelProduct.inputs.amount);
+      const productQuantity = parseInt($productQuantityInput.val(), 10);
+      if (productQuantity <= 0) {
+        $productAmount.val(0);
+        this.updateVoucherRefund();
+
+        return;
+      }
+      const priceFieldName = this.isTaxIncluded ? 'productPriceTaxIncl' : 'productPriceTaxExcl';
+      const productUnitPrice = parseFloat($productQuantityInput.data(priceFieldName));
+      const amountRefundable = parseFloat($productQuantityInput.data('amountRefundable'));
+      const guessedAmount = productUnitPrice * productQuantity < amountRefundable ?
```

Comment:
Make it more readable
```suggestion
      const guessedAmount = (productUnitPrice * productQuantity) < amountRefundable ?
```

Reply:
Done